engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala

```diff
@@ -107,6 +112,26 @@ class FlinkRestManager(config: FlinkConfig, modelData: ModelData, mainClassName:
     }
   }
 
+  override def cancel(processName: ProcessName, user: User): Future[Unit] = withJobOverview(processName)(
+    whenNone = {
+      logger.warn(s"Trying to cancel ${processName.value} which is not present in Flink")
+      Future.successful(())
+    },
+    whenDuplicates = { _ =>
+      //TODO cancel all these jobs
+      logger.warn(s"Trying to cancel ${processName.value} which maps to multiple jobs running in Flink")
+      Future.successful(())
+    },
+    whenSingle = overview => {
+      val status = mapJobStatus(overview)
+      if (processStateDefinitionManager.statusActions(status).contains(ProcessActionType.Cancel) && isNotFinished(overview)) {
```

Comment:
Let's move this if logic to `override protected def cancel(deploymentId: ExternalDeploymentId): Future[Unit]` ?

Reply:
Please see the code now

Reply:
Great!